homeassistant/components/backup/manager.py

```diff
@@ -0,0 +1,157 @@
+"""Backup manager for the Backup integration."""
+from __future__ import annotations
+
+from dataclasses import asdict, dataclass
+import hashlib
+import json
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from securetar import SecureTarFile, atomic_contents_add
+
+from homeassistant.const import __version__ as HAVERSION
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.util import dt, json as json_util
+
+from .const import EXCLUDE_FROM_BACKUP, LOGGER
+
+
+@dataclass
+class Backup:
+    """Backup class."""
+
+    slug: str
+    name: str
+    date: str
+    path: Path
+    size: float
+
+    def as_dict(self) -> dict:
+        """Return a dict representation of this backup."""
+        return {**asdict(self), "path": self.path.as_posix()}
+
+
+class BackupManager:
+    """Backup manager for the Backup integration."""
+
+    _backups: dict[str, Backup] = {}
+    _loaded = False
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the backup manager."""
+        self.hass = hass
+        self.backup_dir = Path(hass.config.path("backups"))
+        self.backing_up = False
+
+    async def load_backups(self) -> None:
+        """Load data of stored backup files."""
+        backups = {}
+
+        def _read_backups() -> None:
+            for backup_path in self.backup_dir.glob("*.tar"):
+                with SecureTarFile(backup_path, "r", gzip=False) as backup_file:
+                    if data_file := backup_file.extractfile("./backup.json"):
+                        data = json.loads(data_file.read())
+                        backup = Backup(
```

Comment:
If an exception happens reading 1 backup, we currently don't recover. Should we log the exception and continue to next backup?